File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

//Distance function calculates the hamming distance
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("the length of the strings must be equal")
	}

	a1 := []rune(a)
	b1 := []rune(b)
	count := 0
	for i := 0; i < len(a1); i++ {
		if a1[i] != b1[i] {
			count++
		}
	}
	return count, nil
}

File: java/hamming/src/main/java/Hamming.java
class Hamming {
	String leftStrand, rightStrand;

    Hamming(String leftStrand, String rightStrand) {
    	this.leftStrand = leftStrand;
    	this.rightStrand = rightStrand;
    	validateInputs(leftStrand, rightStrand);
    }

    void validateInputs(String leftStrand, String rightStrand){
    	if( leftStrand.length() != rightStrand.length())
    		throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
    }

    int getHammingDistance() {
    	int hammingDistance = 0;
    	for(int i =  0; i < leftStrand.length(); i++){
    		if(leftStrand.charAt(i) != rightStrand.charAt(i))
    			hammingDistance++;
    	}
    	return hammingDistance;    
    }

}
